verifier/src/evidence.rs

```diff
@@ -432,4 +456,173 @@ mod test {
 
         assert_matches!(Evidence::new(quote, collateral), Err(Error::Serde(_)));
     }
+
+    struct TestDoubleChainVerifier {
+        failed_certificate_subject: String,
+        error: Error,
+    }
```

Comment:
The `TestDobuleChainVerifier` feels a bit hefty, up for suggestions on minimizing it if possible. I think it's the cost of testing a dependency injection interface.